### ISSUE TITLE
Chromium extension that uses native messaging does not work

### ISSUE THREAD

**Issue by geekware** [NixOS/nixpkgs#85342]
**Describe the bug**
[Plasma Integration](https://chrome.google.com/webstore/detail/plasma-integration/cimiefiiaegbelhefglklhhakcgmhkai) addon in chromium does not work.

After install addon show errors: "**Failed to connect to the main process.**"

"_Verify the installation of the plasma-browser-integration package. The running environment of the Plasma desktop environment must be at least version 5.13._"

"**Specified native messaging host not found.**"

Manually run `plasma-browser-integration-host` from console not help.

**To Reproduce**
1.  Add to `configuration.nix`:
```
{ config, pkgs, lib, ... }:

with lib;

{
environment.systemPackages = (with pkgs; [
    chromium
  ]);
  programs.chromium = {
    enable = true;
    extensions = [
      "cimiefiiaegbelhefglklhhakcgmhkai" # Plasma integration
    ];
  };
}
```
2. `nix-channel --update nixos; nixos-rebuild switch`
3. run `chromium`

**Expected behavior**
No errors.

**Additional context**
I found issue #47340 Everything looks like the same problem but in chromium

**Notify maintainers**

**Metadata**
Please run nix-shell -p nix-info --run "nix-info -m" and paste the result.

 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.31, NixOS, 20.09pre-git (Nightingale)`
 - multi-user?: `no`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.4`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

**Comment by danielfullmer** [comment]
Try adding the following line to your configuration:

```nix
{
  environment.etc."chromium/native-messaging-hosts/org.kde.plasma.browser_integration.json".source = "${pkgs.plasma-browser-integration}/etc/chromium/native-messaging-hosts/org.kde.plasma.browser_integration.json";
}
```

**Comment by geekware** [comment]
> Try adding the following line to your configuration:
> 
> ```nix
> {
>   environment.etc."chromium/native-messaging-hosts/org.kde.plasma.browser_integration.json".source = "${pkgs.plasma-browser-integration}/etc/chromium/native-messaging-hosts/org.kde.plasma.browser_integration.json";
> }
> ```

@danielfullmer now

```
ls /etc/static/chromium/native-messaging-hosts
org.kde.plasma.browser_integration.json

file /etc/static/chromium/native-messaging-hosts/org.kde.plasma.browser_integration.json /etc/static/chromium/native-messaging-hosts/org.kde.plasma.browser_integration.json: symbolic link to /nix/store/cya49vwn2h3jvgai5wbg0bwa3hzm8lf5-plasma-browser-integration-5.17.5/etc/chromium/native-messaging-hosts/org.kde.plasma.browser_integration.json
```
And also does not work. Maybe I need to reboot?

**Comment by worldofpeace** [comment]
> I found issue #47340 Everything looks like the same problem but in chromium

That issue is actually unrelated, it has to do with a patch only in our source build of firefox.

**Comment by geekware** [comment]
I make test with:

```
{ config, pkgs, lib, ... }:

with lib;

{
environment.systemPackages = (with pkgs; [
    chromium
    xdg-desktop-portal-kde
    plasma-browser-integration
    plasma-integration
  ]);

  programs.chromium = {
    enable = true;
    extensions = [
      "cimiefiiaegbelhefglklhhakcgmhkai" # Plasma integration
    ];
  };

  environment.etc."chromium/native-messaging-hosts/org.kde.plasma.browser_integration.json".source = "${pkgs.plasma-browser-integration}/etc/chromium/native-messaging-hosts/org.kde.plasma.browser_integration.json";

}

```

in `configuration.nix`.

The same error.

**Comment by danielfullmer** [comment]
I can't reproduce this issue. I just tested in a vm using `nixos-rebuild build-vm` with this configuration:
```
{ config, pkgs, lib, ... }:

with lib;
{
  environment.systemPackages = (with pkgs; [ chromium  ]);

  programs.chromium = {
    enable = true;
    extensions = [
      "cimiefiiaegbelhefglklhhakcgmhkai" # Plasma integration
    ];
  };

  environment.etc."chromium/native-messaging-hosts/org.kde.plasma.browser_integration.json".source = "${pkgs.plasma-browser-integration}/etc/chromium/native-messaging-hosts/org.kde.plasma.browser_integration.json";

  users.users.danielrf = {
    isNormalUser = true;
    password = "changeme";
  };

  services.xserver.enable = true;
  services.xserver.desktopManager.plasma5.enable = true;
  services.xserver.displayManager.lightdm.enable = true;

  virtualisation.memorySize = 2048;
  virtualisation.diskSize = 5000;
}
```

Are you using plasma5?

**Comment by geekware** [comment]
Yes plasma5. Looks like it worked for me too. Now all errors are gone. Thanks @danielfullmer !

**Comment by worldofpeace** [comment]
Why are we adding `chromium` to systemPackages when using the chromium program module? It does this for you.

**Comment by danielfullmer** [comment]
Just an oversight on my part. You're right, adding chromium to systemPackages is not necessary.

**Comment by haizaar** [comment]
Thanks @geekware - the fix worked for me!

Can you please comment on what's the role of `plasma-integration` package that you install?

**Comment by SuperSandro2000** [comment]
Proposed fix in #232528
